.travis.yml

```diff
@@ -1,13 +1,56 @@
-sudo: required
+sudo: false
 
-services:
-  - docker
+cache:
+  directories:
+    - $HOME/.composer/cache
 
-before_script:
-  - docker-compose -f docker-compose.yml up --build --force-recreate -d
-  - sh ./travis/connection_check.sh elastic 9200
-  - sh ./travis/connection_check.sh mysql 3306
-  - sh ./travis/dependencies_check.sh
+language: php
+
+env:
+  global:
+    - COVERAGE=0
+
+matrix:
+  include:
+    - php: 7.1
+      env: LARAVEL='5.5.*'
+    - php: 7.1
+      env: LARAVEL='5.6.*'
+    - php: 7.1
+      env: LARAVEL='5.7.*'
+    - php: 7.1
+      env: LARAVEL='5.8.*'
+    - php: 7.2
+      env: LARAVEL='5.5.*'
+    - php: 7.2
+      env: LARAVEL='5.6.*'
+    - php: 7.2
+      env: LARAVEL='5.7.*'
+    - php: 7.2
+      env: LARAVEL='5.8.*'
+    - php: 7.3
+      env: LARAVEL='5.5.*'
+    - php: 7.3
+      env: LARAVEL='5.6.*'
+    - php: 7.3
+      env: LARAVEL='5.7.*'
+    - php: 7.3
+      env: COVERAGE=1 LARAVEL='5.8.*'
+  fast_finish: true
+
+before_install:
+  - phpenv config-rm xdebug.ini || true
+  - echo "memory_limit=4G" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
+  - travis_retry composer self-update
+  - travis_retry composer global require hirak/prestissimo --update-no-dev
+  - travis_retry composer require "illuminate/contracts:${LARAVEL}" --no-interaction --no-update
+
+install:
+  - travis_retry composer install --prefer-dist --no-interaction --no-suggest
 
 script:
-  - docker-compose -f docker-compose.yml exec php /app/driver/vendor/bin/phpunit -c /app/driver/phpunit.xml.dist
+  - if [[ $COVERAGE = 1 ]]; then phpdbg -qrr vendor/bin/phpunit --coverage-clover=coverage.xml; fi
+  - if [[ $COVERAGE = 0 ]]; then vendor/bin/phpunit --colors=always --verbose; fi
+
+#after_success:
+#  - if [[ $COVERAGE = 1 ]]; then bash <(curl -s https://codecov.io/bash); fi
```

Comment:
You might want to enable this service @babenkoivan ?
This is free for public repositories.